core-backend/sandbox/src/env.rs

```diff
@@ -173,19 +165,39 @@ where
         env_builder.add_host_func("env", "free", Funcs::free);
         env_builder.add_host_func("env", "gas", Funcs::gas);
 
-        let mut memory_wrap = MemoryWrap::new(mem.clone());
+        let memory_wrap = MemoryWrap::new(mem.clone());
         let mut runtime = Runtime {
-            ext: &mut ext,
-            memory: &mem,
-            memory_wrap: &mut memory_wrap,
+            ext,
+            memory: mem,
+            memory_wrap,
             err: FuncError::Terminated(TerminationReason::Success),
         };
 
-        let mut instance = match Instance::new(binary, &env_builder, &mut runtime) {
+        let instance = match Instance::new(binary, &env_builder, &mut runtime) {
             Ok(inst) => inst,
             Err(e) => return Err((runtime.ext.gas_amount(), ModuleInstantiation(e)).into()),
         };
 
+        Ok(Self { instance, runtime })
```

Comment:
```suggestion
        match Instance::new(binary, &env_builder, &mut runtime) {
            Ok(instance) => Ok(Self { instance, runtime }),
            Err(e) => Err((runtime.ext.gas_amount(), ModuleInstantiation(e)).into()),
        }
```

Probably `map` and `map_err` will also be possible to use despite the compiler strange stuff